task/task_service.go

```diff
@@ -10,67 +10,50 @@ import (
 // Action represents any work to be done when executing a task.
 type Action func() error
 
-// TaskObject represents a task that is executable.
-type TaskObject struct {
```

Comment:
the file was there and it felt logic for me to put the type `Task` into the file `task.go`. Since you ask, I will revert. there is no need for this change by itself.